tests/org.eclipse.reddeer.eclipse.test/src/org/eclipse/reddeer/eclipse/test/integration/AvailableSoftwareSitesPreferencePageTest.java

```diff
@@ -0,0 +1,211 @@
+		
+		dialog.ok();
+		
+	}
+	
+	@Before
+	public void openAvailableSoftwareSitesPreferencePage() {
+		dialog = new WorkbenchPreferenceDialog();
+		dialog.open();
+		page = new AvailableSoftwareSitesPreferencePage(dialog);
+		dialog.select(page);
+	}
+	
+	@Test
+	public void pageIsOpened() {
+		assertTrue("Page does not have right name", page.getName().contains("Available Software Sites"));
+	}
+	
+	@Test
+	public void getItemsTest() {
+		assertFalse(page.getItems().isEmpty());
+	}
+	
+	@Test
+	public void getItemTest() {
+		String itemText = page.getItem(ECLIPSE).getText();
+		assertTrue("Item has text " + itemText, page.getItem(ECLIPSE).getText().contains(ECLIPSE));
+	}
+	
+	@Test
+	public void getItemAlternateSearchTest() {
+		String itemText = page.getItem(REDDEER_URL, true).getText(1);
+		assertTrue("the item has text " + itemText, itemText.contains(REDDEER_URL));
+		itemText = page.getItem(ECLIPSE, false).getText();
+		assertTrue("the item has text " + itemText, itemText.contains(ECLIPSE));
+		TableItem item = page.getItem(ECLIPSE, true);
+		assertNull("Item is not null", item);
+		
+	}
+	
+	@Test
+	public void toggleAllItemsTest() {
+		page.toggleAllItems(false);
+		for(TableItem item : page.getItems()) {
+			assertFalse(item.isChecked());
+		}
+		page.toggleAllItems(true);
+		for(TableItem item : page.getItems()) {
+			assertTrue(item.isChecked());
+		}
+	}
+	
+	@Test
+	public void toggleItemTest() {
+		page.toggleItem(ECLIPSE, false);
+		assertFalse(page.getItem(ECLIPSE).isChecked());
+		page.toggleItem(ECLIPSE, true);
+		assertTrue(page.getItem(ECLIPSE).isChecked());
+	}
+	
+	@Test
+	public void toggleItemAlternativeSearchTest() {
+		page.toggleItem(REDDEER_URL, false, true);
+		assertFalse(page.getItem(REDDEER_URL, true).isChecked());
+		page.toggleItem(REDDEER_URL, true, true);
+		assertTrue(page.getItem(REDDEER).isChecked());
+	}
+	
+	@Test
+	public void selectItemTest() {
+		page.selectItem(REDDEER);
+		assertTrue(page.isButtonEnabled("Edit"));
+	}
+	
+	@Test
+	public void selectItemAlternativeSearchTest() {
+		page.selectItem(ECLIPSE_URL, true);
+		assertTrue(page.isButtonEnabled("Edit"));
+	}
+	
+	@Test
+	public void isItemEnabledTest() {
+		assertTrue(page.isItemEnabled(ECLIPSE));
+		page.toggleItem(ECLIPSE, false);
+		assertFalse(page.isItemEnabled(ECLIPSE));
+	}
+	
+	@Test
+	public void isItemEnabledAlternativeSearchTest() {
+		assertTrue(page.isItemEnabled(ECLIPSE_URL, true));
+		page.toggleItem(ECLIPSE, false);
+		assertFalse(page.isItemEnabled(ECLIPSE_URL, true));
+	}
+
+	@Test
+	public void isButtonEnabledTest() {
+		assertTrue(page.isButtonEnabled("Add..."));
+		assertFalse(page.isButtonEnabled("Edit"));
+	}
+	
+	@Test
+	public void clickAddTest() {
+		page.clickAdd();
+		shell = new DefaultShell();
+		assertTrue("Shell with title Add Site is not opened", shell.getText().contains("Add Site"));
+	}
+	
+	@Test
+	public void clickEditTest() {
+		page.selectItem(ECLIPSE);
+		page.clickEdit();
+		shell = new DefaultShell();
```

Comment:
Here could be better fit to use shell's name (Add Site). Same for the other test cases as well: Edit Sites, Remove Sites, etc...